### PR TITLE
feat: basic configure for eslint

### PR DESCRIPTION
**Pull request by loc-nguyenthien**
Add basic configurations for eslint in our base project, with following features:

- [x] [Require] - Import first
- [x] [Require] - Enforces Rules of Hooks
- [x] [Require] - Duplicate import
- [x] [Require] - Arrow body style
- [x] [Warning] - Check cycle import

### CODE REVIEW

File: template/.eslintrc.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
   root: true,
-  extends: '@react-native-community',
+  extends: ['@react-native-community', 'plugin:react/recommended', 'plugin:react-hooks/recommended'],
   plugins: ['unused-imports'],
   rules: {
     semi: 'off',
     'comma-dangle': 'off',
     'no-unused-vars': 'error',
     'unused-imports/no-unused-imports': 'error',
+    'import/no-cycle': 'error',
```
[the remark this answers is not in the document]
Comment (honghoangsts) [reply template/.eslintrc.js:10]: Yes. There are some case we can not fix cycle import.

File: template/.eslintrc.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+    'import/first': 'error',
+    'import/no-duplicates': 'warn',
+    'arrow-body-style': ['warn', 'always'],
```
Comment (honghoangsts) [inline template/.eslintrc.js:13]: This one should be `error`

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: template/.eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (loc-nguyenthien) [reply template/.eslintrc.js]: @honghoangsts I have updated accordingly

File: template/.eslintrc.js (hunk 2 above)
Comment (loc-nguyenthien) [reply template/.eslintrc.js]: @honghoangsts I have updated accordingly

**Review by vunguyendhSTS — APPROVED** [review]

**Review by loido — APPROVED** [review]

**Review by tuledu — APPROVED** [review]

**Review by baonguyenhsts — APPROVED** [review]

File: template/.eslintrc.js (hunk 2 above)
Comment (honghoangsts) [reply template/.eslintrc.js]: `import/no-duplicates` => should be error as well

Comment (loc-nguyenthien) [reply template/.eslintrc.js]: @honghoangsts updated accordingly

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**